@Test(timeout = DEFAULT_TEST_TIMEOUT)
public void testRunNow_ChildDiamond_StopOnError_2Fails() throws InterruptedException {
    String group = "diamondTest";
    Task task1 = new Task("task1", group, BEAN_WAIT_JOB, true, true, "3000", "");
    taskManager.save(task1);
    // set task 2 to fail
    Task task2 = new Task("task2", group, "failJob", true, true, "true", "");
    taskManager.save(task2);
    Task task3 = new Task("task3", group, BEAN_WAIT_JOB, true, true, "5000", "");
    taskManager.save(task3);
    Task task4 = new Task("task4", group, BEAN_WAIT_JOB, true, true, "1000", "");
    taskManager.save(task4);
    // now create our task "diamond"
    task1.addChildTask(task2);
    task1.addChildTask(task3);
    taskManager.save(task1);
    task2.addChildTask(task4);
    taskManager.save(task2);
    task3.addChildTask(task4);
    taskManager.save(task3);
    schedulerManager.runTaskNow(task1);
    // task 4 should run
    waitForTask(task4);
    List<TaskRun> taskRuns = taskRunManager.findByTaskId(task2.getId());
    assertEquals(1, taskRuns.size());
    TaskRun taskRun = taskRuns.get(0);
    assertEquals(Status.FAILED, taskRun.getStatus());
    taskRuns = taskRunManager.findByTaskId(task4.getId());
    assertEquals(1, taskRuns.size());
    taskRun = taskRuns.get(0);
    assertEquals(Status.SUCCESS, taskRun.getStatus());
}
